tools/devservices_healthcheck.py

```diff
@@ -1,47 +1,183 @@
 from __future__ import annotations
 
+import argparse
 import os
 import subprocess
 import time
+from collections.abc import Sequence
+from typing import Callable
 
 
-def run_cmd(
-    args: list[str],
-    *,
-    retries: int = 3,
-    timeout: int = 5,
-) -> None:
-    for retry in range(1, retries + 1):
-        returncode = subprocess.call(args)
+class HealthcheckError(Exception):
+    pass
+
+
+class HealthCheck:
+    id: str
+    container_name: str
+    check_by_default: bool
+    check: Callable[[], None]
+    deps: list[str]
+    retries: int
+    timeout_secs: int
```

Comment:
these should not be specified here -- these should be the argument types of the `__init__` function

please get this file passing again with `mypy --disallow-untyped-defs tools tests/tools`

Reply:
Fixed the mypy types, is this something that will be added to CI after your typing changes?

Reply:
eventually we'll ratchet it up everywhere yeah -- but for now we can hold ourselves to a higher standard